stdlib/asm/math/u64.masm

```diff
@@ -29,13 +29,42 @@ export.add_unsafe
     drop
 end
 
+# Performs addition of two unsigned 64 bit integers, fails when overflowing. #
+# The input values are assumed to be represented using 32 bit limbs, fails if they are not. #
+# Stack transition looks as follows: #
+# [b_hi, b_lo, a_hi, a_lo, ...] -> [c_hi, c_lo, ...], where c = (a + b) % 2^64 #
+export.checked_add
+    exec.u32assert4
+    swap
+    movup.3
+    u32add.unsafe
+    movup.3
+    movup.3
+    u32addc.unsafe
+    eq.0
+    assert
+end
+
+# Performs addition of two unsigned 64 bit integers preserving the overflow. #
+# The input values are assumed to be represented using 32 bit limbs, but this is not checked. #
+# Stack transition looks as follows: #
+# [b_hi, b_lo, a_hi, a_lo, ...] -> [overflowing_flag, c_hi, c_lo, ...], where c = (a + b) % 2^64 #
+export.overflowing_add
+    swap
+    movup.3
+    u32add.unsafe
+    movup.3
+    movup.3
+    u32addc.unsafe
+end
+
```

Comment:
I think we should be able to implement `checked_add` and `wrapping_add` by relying on `overflowing_add`. For example, `wrapping_add` could be just:

```
exec.overflowing_add
drop
```